### ISSUE TITLE
Build Blog

### ISSUE THREAD

**Issue by gvn** [mozilla/science.mozilla.org#125]
P1

- [x] Post list
- [x] Post detail

P2

- [x] Pagination for list view 
- [ ] Comments (Disqus) **Blocked**
- [ ] Pagination for detail view **Blocked by WP API missing data**

P3

- [ ] Post list: Search
- [ ] Search results view
- [ ] Post list : Filter results

**API:** http://wp.mozillascience.org/wp-json

**Current Node implementation:** https://github.com/mozilla/science.mozilla.org/blob/mozillascience.org/routes/blog.js

**Comment by gvn** [comment]
@hannahkane Were we planning on doing search for R1? I'd like to deprioritize or move it to R2 if possible?

**Comment by gvn** [comment]
Consuming this API: http://wp.mozillascience.org/wp-json

**Comment by hannahkane** [comment]
Do: list view with 10 posts, detail view 
P2: comments, pagination 
Drop: search, filter, next/prev on detail

**Comment by kristinashu** [comment]
On the current site, an author's name links to their profile page. We will need to temporarily disable those links since we are not doing members right now.

**Comment by gvn** [comment]
I think we might have to upgrade the WP REST API plugin to V2 to achieve (sane/hack-free) client-side pagination.

Currently the `X-WP-Total` header isn't whitelisted by CORS, so I can't read it to see how many pages of posts there are. It doesn't seem like there's any way to enable `Access-Control-Expose-Headers` for `X-WP-Total` via the plugin's web interface. I'm not sure if there's server-side config, but my guess is "no".

Options:

1. Switch to the V2 API now. This might break the current blog implementation and isn't guaranteed to solve the CORS issue.
2. Set up a middleware agent to hit the WP API and expose the pagination data safely for CORS. This is hacky, and I'd prefer not to do it.
3. See if there's any way to run V1 and V2 of the API simultaneously. I don't think there is as they can't coexist on a single WP instance. V1 takes precedence.
4. Wait until R2 to implement pagination, preferably by updating the API to V2.

**TL;DR** – Pagination is turning into a bit of a black hole. Recommend we push it to R2 when we can more safely switch over to the V2 API.

CC: @acabunoc @hannahkane ^

**Comment by gvn** [comment]
Actually, one more option occurred to me. I can keep allowing additional pages of posts to be requested until the request just breaks. It's not very elegant, but I think it will work...

**Comment by hannahkane** [comment]
@gvn - can you describe what the alternative to pagination is? Is it all posts on one page?

**Comment by gvn** [comment]
Well, right now we have 10 posts with no way to get to more.

The option I'm suggesting is that you can load everything but you might still get a LOAD MORE button if there's nothing left. It's kind of an edge case though.

**Comment by hannahkane** [comment]
I think it would be preferable to give users the option to load more, even if it eventually breaks. What do you think, @acabunoc @kristinashu ?

**Comment by gvn** [comment]
Agreed. I am running into other problems with the V1 API (#338), so upgrading to V2 might be inevitable.

**Comment by acabunoc** [comment]
I got pagination working the current site by adding `?page=`. Not sure if the same will work for you?
https://github.com/mozilla/science.mozilla.org/blob/mozillascience.org/routes/blog.js#L16

We didn't have pagination for a year on the old site. I'm good with pushing to phase 2 if it comes to it.

**Comment by gvn** [comment]
To clarify, I can paginate. The issue is that I can't tell how many pages there are, so I'd have to blindy keep requesting more pages until things fail. I think that's ok though.

**Comment by acabunoc** [comment]
Ah, yeah that's cool

**Comment by gvn** [comment]
Moving to `Phase 2` per standup discussion.

**Comment by hannahkane** [comment]
I'm going to close this ticket, since we're opening up individual tickets for the next blog iteration.
